static/js/compose.js

```diff
@@ -270,13 +272,46 @@ export function enter_with_preview_open() {
     }
 }
 
+function check_unsubscribed_stream_for_send(stream_name, autosubscribe) {
+    let result;
+    if (!autosubscribe) {
+        return "not-subscribed";
+    }
+
+    // In the rare circumstance of the autosubscribe option, we
+    // *Synchronously* try to subscribe to the stream before sending
+    // the message.  This is deprecated and we hope to remove it; see
+    // #4650.
+    channel.post({
+        url: "/json/subscriptions/exists",
+        data: {stream: stream_name, autosubscribe: true},
+        async: false,
+        success(data) {
+            if (data.subscribed) {
+                result = "subscribed";
+            } else {
+                result = "not-subscribed";
+            }
+        },
+        error(xhr) {
+            if (xhr.status === 404) {
+                result = "does-not-exist";
+            } else {
+                result = "error";
+            }
+        },
+    });
+    return result;
+}
```

Comment:
Why did you move this function to a different file? I don't see how it's constructive, since we generally do have a code structure goal of keeping validation logic for the compose box in `compose_validate.js`.